backend/app/src/main/java/com/moh/phlat/backend/service/DbUtilityServiceImpl.java

```diff
@@ -202,17 +207,36 @@ public void loadProcessDataToPlr(Long controlTableId, String authenticatedUserId
 			Iterable<ProcessData> processDataList = processDataRepository
 					.getAllProcessDataByControlTableId(controlTableId);
 
-			for (ProcessData rec : processDataList) {
-				// skip record marked as DO_NOT_LOAD and send to PLR VALID records only
-				if (!"Y".equals(rec.getDoNotLoadFlag()) && RowStatusService.VALID.equals(rec.getRowstatusCode())) {
-					logger.info("loading process data with id: {} to PLR.", rec.getId());
-
-					// TO-DO call to PLR via ESB here
-					// loadPlrViaEsb(control, s);
+			if (plrDataLoad.getPlrEsbBoundary() != null && plrDataLoad.getPlrEsbBoundary().isReadyToConnect()) {
+				for (ProcessData rec : processDataList) {
+					// skip record marked as DO_NOT_LOAD and send to PLR VALID records only
+					if (!"Y".equals(rec.getDoNotLoadFlag()) && RowStatusService.VALID.equals(rec.getRowstatusCode())) {
+						logger.info("loading process data with id: {} to PLR.", rec.getId());
+
+						MaintainResults loadResults = plrDataLoad.loadPlrViaEsb(control, rec);
+						for (PlrError plrError : loadResults.getFacilityResult().getPlrErrors()) {
+							Message msg = Message.builder()
+									 .messageType(plrError.getErrorType())
```

Comment:
I don't think the PLR to PHLAT error code mapping has been done yet, and this was a piece being developed in Stage B so there might be some outdated elements to it. I think for now, because PHLAT will mainly be used by CGI on its first release and because our main focus is to get the load functionality working, we should be fine to put the PLR codes in where the mapped codes were planned to be; then, if we do get to mapping them out to new PHLAT-specific codes, I can simply change which columns the PLR codes are going into in the Message table.

Reply:
ok